crates/core/tedge_api/src/workflow/supervisor.rs

```diff
@@ -136,10 +136,10 @@ impl WorkflowSupervisor {
     ///
     /// Return the current version if any.
     pub fn use_current_version(&mut self, operation: &OperationName) -> Option<WorkflowVersion> {
-        match self.workflows.get_mut(&operation.as_str().into()) {
-            Some(versions) => versions.use_current_version().cloned(),
-            None => None,
-        }
+        self.workflows
+            .get_mut(&operation.as_str().into())
+            .map(WorkflowVersions::use_current_version)?
+            .cloned()
```

Comment:
This function could also take `&str`, and we could also implement `Borrow<str>` for `OperationType`, which would save us the conversion into `&OperationType` when we query `self.workflows`.

The `.map()?` is also an interesting pattern. I think either of these are clearer:
```suggestion
        self.workflows
            .get_mut(&operation.as_str().into())
            .and_then(WorkflowVersions::use_current_version)
            .cloned()
```

```suggestion
        self.workflows
            .get_mut(&operation.as_str().into())?
            .use_current_version()
            .cloned()
```

Reply:
I opted for the 2nd proposal: https://github.com/thin-edge/thin-edge.io/pull/3309/commits/eecd22274eb3baed62ffefa53be54d1768dc1c99

Reply:
> This function could also take &str, and we could also implement Borrow<str> for OperationType, which would save us the conversion into &OperationType when we query self.workflows.

`Borrow<str>` cannot be implemented neither by `OperationType` (which is an enum with no string carrier) nor by `OperationName` (which is an alias).